howso/engine/session.py

```diff
@@ -184,22 +181,22 @@ def set_metadata(self, metadata: Optional[Dict[str, Any]]) -> None:
         self._metadata = metadata
         self._update()
 
-    def _update_attributes(self, session: BaseSession) -> None:
+    def _update_attributes(self, session) -> None:
         """
         Update the protected attributes of the session.
 
         Parameters
         ----------
-        session : BaseSession
+        session : Session
             The base session instance.
 
         Returns
         -------
         None
         """
-        for key in self.attribute_map.keys():
+        for key in ATTRIBUTE_MAP.keys():
```

Comment:
Shouldn't be using this global attribute map here. Let just define it on the Session class.

Reply:
Addressed by: 2ba04a5